src/gameState.ts

```diff
@@ -0,0 +1,82 @@
+import { Board } from "./board/board";
+import { loadGameConfigFromFile } from "./gameLoader";
+import { Coordinates, checkIfMoveIsAllowed } from "./move/move";
+import { Player } from "./player";
+import { Tile } from "./tile/tile";
+
+const initialGameState: GameState = {
+  currentPlayer: "black",
+  turnNumber: 1,
+};
+
+export interface GameState {
+  board?: Board;
+  turnNumber: number;
+  currentPlayer: Player;
+  winner?: Player;
+  selectedMoveCordinates?: Coordinates;
+}
+export type GameActionName =
+  | "SELECT_TILE"
+  | "FORFEIT"
+  | "WAITING"
+  | "END_TURN"
+  | "LOAD_BOARD";
+export interface GameStateAction {
+  name: GameActionName;
+  selectedMoveCoordinates?: Coordinates;
+  newBoard?: Board;
+}
+
+const setAllowedTilesOnBoard = (gameState: GameState): Board => {
+  if (gameState.board == null) {
+    return null;
+  }
+
+  if (gameState.turnNumber === 1) {
+    let newBoard: Board = JSON.parse(JSON.stringify(gameState.board));
+    gameState.board.forEach((line, y) =>
+      line.forEach((tile, x) => {
+        const currentTile = newBoard[y][x];
+
+        if (currentTile == null) {
+          return;
+        }
+
+        newBoard[y][x].moveAllowed = checkIfMoveIsAllowed(
+          { x, y },
+          gameState.turnNumber === 1
+        );
+      })
+    );
+    return newBoard;
+  }
+};
+
+export const computeGameState = (
+  gameState: GameState | undefined,
```

Comment:
Why do you allow `gameState` to be undefined?

Also, you never check that  `gameState` is actually defined later in the code.

Reply:
No longer exists since merge with #8